pennylane/transforms/sign_expand.py

```diff
@@ -0,0 +1,327 @@
+      phis (List[float]): Optimal phi values for the QSP part associated with the respective delta and J
+
+    Returns:
+      tapes (List[qml.tape]): Expanded tapes from the original tape that measures the terms via the approximate sgn decomposition
+    """
+    coeffs = hamiltonian.data
+    tapes = []
+    for mu, time in zip(mus, times):
+        with tape.__class__() as new_tape:
+            # Put state prep and ansatz on tape in the 'old' register
+            for op in tape.operations:
+                op.queue()
+
+            # Put QSP and Hadamard test on the two ancillas Target and Control
+            qml.Hadamard("Hadamard")
+            for i, phi in enumerate(phis):
+                qml.CRX(phi, wires=["Hadamard", "Target"])
+                if i == len(phis) - 1:
+                    qml.CRY(np.pi, wires=["Hadamard", "Target"])
+                else:
+                    evolve_under(hamiltonian.ops, coeffs, 2 * time)
+                    qml.CRZ(-2 * mu * time, wires=["Hadamard", "Target"])
+            qml.Hadamard("Hadamard")
+
+            if tape.measurements[0].return_type == qml.measurements.Expectation:
+                qml.expval(-1 * qml.PauliZ("Hadamard"))
+            else:
+                qml.var(qml.PauliZ("Hadamard"))
+
+        tapes.append(new_tape)
+    return tapes
+
+
+def sign_expand(tape, circuit=False, J=10, delta=0.0):
+    r"""
+    Splits a tape measuring a (fast-forwardable) Hamiltonian expectation into mutliple tapes of the Xi or sgn decomposition,
+    and provides a function to recombine the results.
+
+    Implementation of ideas from arXiv:2207.09479
+
+    For the calculation of variances, one assumes an even distribution of shots among the groups as an adaptive shot allocation onto tapes is not implemented in Pennylane yet.
+
+    Args:
+        tape (.QuantumTape): the tape used when calculating the expectation value
+            of the Hamiltonian
+        circuit (bool): Toggle the calculation of the analytical Xi decomposition or if True constructs the circuits of the approximate sign decomposition
+            to measure the expectation value
+        J (int): The times the time evolution of the hamiltonian is repeated in the quantum signal processing approximation of the sgn-decomposition
+        delta (float): The minimal
+    Returns:
+        tuple[list[.QuantumTape], function]: Returns a tuple containing a list of
+        quantum tapes to be evaluated, and a function to be applied to these
+        tape executions to compute the expectation value.
+
+    **Example**
+
+    Given a Hamiltonian,
+
+    .. code-block:: python3
+
+        H = qml.PauliZ(0) + 0.5 * qml.PauliZ(2) + qml.PauliZ(1)
+
+    and a tape of the form,
+
+    .. code-block:: python3
+
+        with qml.tape.QuantumTape() as tape:
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PauliX(wires=2)
+
+            qml.expval(H)
```

Comment:
We want the users to use quantum functions instead of tapes. Could you please change the examples?

```python
@qml.qnode(dev)
def circuit():
    qml.Hadamard(wires=0)
    qml.CNOT(wires=[0, 1])
    qml.PauliX(wires=2)
    return qml.expval(H)
```